presto-main/src/main/java/com/facebook/presto/server/remotetask/HttpRemoteTaskFactory.java

```diff
@@ -165,4 +181,63 @@ public RemoteTask createRemoteTask(Session session,
                 stats,
                 isBinaryTransportEnabled);
     }
+
+    @Override
+    public void removeRemoteSource(TaskId taskId, TaskId remoteSourceTaskId, Consumer<PrestoException> onFailure)
+    {
+        // TODO: Check whether we need the onFailure callback. Since this method is used for recoverable execution,
+        //  and if the cleanup failed, the query will fail.
+        URI remoteSourceUri = uriBuilderFrom(locationFactory.createLocalTaskLocation(taskId))
+                .appendPath("remote-source")
+                .appendPath(remoteSourceTaskId.toString())
+                .build();
+
+        Request request = prepareDelete()
+                .setUri(remoteSourceUri)
+                .build();
+        RequestErrorTracker errorTracker = new RequestErrorTracker(
+                taskId,
+                remoteSourceUri,
+                maxErrorDuration,
+                errorScheduledExecutor,
+                "Remove exchange remote source");
+
+        executeDestroyExchangeSourceRequest(errorTracker, request, onFailure);
+    }
+
+    // Copied from https://github.com/prestodb/presto/commit/36a60a6f6e029ddd890d0eb10de01f339eedfebb#diff-b040516898118558de47053d230b8420
+    private void executeDestroyExchangeSourceRequest(RequestErrorTracker errorTracker, Request request, Consumer<PrestoException> onFailure)
```

Comment:
This method is mainly copied from https://github.com/prestodb/presto/commit/36a60a6f6e029ddd890d0eb10de01f339eedfebb#diff-b040516898118558de47053d230b8420 with some minor adaptions.

Reply:
@arhimondr : Reverted in https://github.com/prestodb/presto/pull/11065 :)